app/src/main/java/org/wikipedia/feed/onthisday/OnThisDayFragment.java

```diff
@@ -88,83 +97,81 @@ public static OnThisDayFragment newInstance(int age, WikiSite wikiSite) {
-                && savedInstanceState == null) {
-            final int animDelay = 500;
-            dayText.postDelayed(() -> {
-                if (!isAdded() || dayText == null) {
-                    return;
-                }
-                updateContents(age);
-                ValueAnimator colorAnimation = ValueAnimator.ofObject(new ArgbEvaluator(), dayText.getCurrentTextColor(),
-                        ResourceUtil.getThemedColor(requireContext(), R.attr.material_theme_primary_color));
-                colorAnimation.addUpdateListener(animator -> {
-                    if (dayText != null) {
-                        dayText.setTextColor((Integer) animator.getAnimatedValue());
-                    }
-                });
-                colorAnimation.start();
-            }, animDelay);
-        } else {
-            dayText.setTextColor(ResourceUtil.getThemedColor(requireContext(), R.attr.material_theme_primary_color));
+        final int animDelay = (requireActivity().getWindow().getSharedElementEnterTransition() != null
+                && savedInstanceState == null) ? 500 : 0;
+        onThisDayTitleView.postDelayed(() -> {
+            if (!isAdded() || onThisDayTitleView == null) {
+                return;
+            }
             updateContents(age);
-        }
+        }, animDelay);
 
-        progressBar.setVisibility(View.GONE);
-        eventsRecycler.setVisibility(View.GONE);
-        errorView.setVisibility(View.GONE);
+        progressBar.setVisibility(GONE);
+        eventsRecycler.setVisibility(GONE);
+        errorView.setVisibility(GONE);
         return view;
     }
 
-    public void onBackPressed() {
-        dayText.setTextColor(ResourceUtil.getThemedColor(requireContext(), R.attr.primary_text_color));
-    }
-
     private void updateContents(int age) {
         Calendar today = DateUtil.getDefaultDateFor(age);
         requestEvents(today.get(Calendar.MONTH), today.get(Calendar.DATE));
         funnel = new OnThisDayFunnel(WikipediaApp.getInstance(), wiki,
                 (InvokeSource) requireActivity().getIntent().getSerializableExtra(INTENT_EXTRA_INVOKE_SOURCE));
     }
 
+    @SuppressWarnings("checkstyle:magicnumber")
     private void requestEvents(int month, int date) {
-        progressBar.setVisibility(View.VISIBLE);
-        eventsRecycler.setVisibility(View.GONE);
-        errorView.setVisibility(View.GONE);
+        progressBar.setVisibility(VISIBLE);
+        eventsRecycler.setVisibility(GONE);
+        errorView.setVisibility(GONE);
 
         disposables.add(ServiceFactory.getRest(wiki).getOnThisDay(month + 1, date)
                 .subscribeOn(Schedulers.io())
                 .observeOn(AndroidSchedulers.mainThread())
-                .doAfterTerminate(() -> progressBar.setVisibility(View.GONE))
+                .doAfterTerminate(() -> {
+                    progressBar.setVisibility(GONE);
+                    eventsRecycler.postDelayed(() -> {
```

Comment:
Adding a check to prevent a crash?

Reply:
@cooltey yes, the 01 checks are to prevent crash.. since we dont have values in all cases for the year

Reply:
I was thinking about adding something like `isAdded` or `isDestoryed`